processing/src/main/java/org/apache/druid/segment/Segment.java

```diff
@@ -64,25 +73,36 @@ public interface Segment extends Closeable
    *
    * @return instance of clazz, or null if the interface is not supported by this segment
    *
-   * @see StorageAdapter storage adapter for queries. Never null.
+   * @see CursorFactory to make cursors to run queries
```

Comment:
yea, though like, in the same way it was true for `StorageAdapter`. It can actually be null in practice because we got the storage adapter through `ReferenceCountingSegment`, which returns `null` for a lot of its methods if the backing segment has been closed due to being dropped, so the javadoc was kind of lies.

Reply:
updated to specify never null. Did not clarify about `ReferenceCountingSegment` because its kind of true for almost every method on `Segment` and wasn't quite sure how to best describe it.